### ISSUE TITLE
Could not find target name 'Unity Editor'. Is it running?

### ISSUE THREAD

**Issue by andronio-mf** [Unity-Technologies/vscode-unity-debug#197]
I am getting the same issue reported [here](https://github.com/Unity-Technologies/vscode-unity-debug/issues/5), but the suggested fix does not solve my problem.

## Expected Behavior
I should be able to debug a unity game

## Current Behavior
I get the error message in the title and cannot debug anything. Debug log:
```
UnityDebug: Initializing
UnityDebug: Searching for Unity process 'Unity Editor'
```

## Possible Solution
Yes, I ensured that "Editor Attaching" is checked.
I tried to wipe and cleanly install `mono-devel` multiple times. I tried to install different versions of mono (6.10, 6.12, and the latest 6.12.0.122).
I also tried to install several old version of the extension but to no avail.

I ran the suggested code in the previous issue, and found that no process named "Unity Editor" is running. These are all my unity processes:
```
Process ID 6062 Name 'Unity.app'
Process ID 6071 Name 'UnityPackageManager'
Process ID 6189 Name 'UnityShaderCompiler'
Process ID 6201 Name 'UnityHelper'
Process ID 16536 Name 'Unity.Licensing.Client'
Process ID 20412 Name 'UnityHub.AppImage'
Process ID 22983 Name 'UnityHub.AppImage'
Process ID 3645 Name 'unity-settings-daemon'
Process ID 3880 Name 'unity-panel-service'
Process ID 4010 Name 'unity-fallback-mount-helper'
Process ID 20403 Name 'unityhub-bin'
Process ID 20427 Name 'unityhub-bin'
Process ID 20429 Name 'unityhub-bin'
Process ID 20463 Name 'unityhub-bin'
Process ID 20471 Name 'unityhub-bin'
Process ID 22993 Name 'unityhub-bin'
Process ID 22995 Name 'unityhub-bin'
Process ID 24989 Name 'unity-scope-home'
Process ID 25001 Name 'unity-files-daemon'
Process ID 25002 Name 'unity-scope-loader'
```

I tried to change my configuration name to match several of these processes (e.g. Unity.app) but the extension cannot find any, despite them being clearly running.

## Context
I am trying to start debugging a unity game. This did work a week ago but suddenly stopped working. I suspect the latest mono-devel update (build 6.12.0.122) may have broken something, but installing old versions is impossible as they produce unsolvable dependency conflicts.

## Your Environment
* Unity Version: 2019.3.8f1
* VS Code Version: 1.53.2
* Version of the debugger tool: 3.0.2
* Operating System and version: Ubuntu 18.04

**Comment by deitry** [comment]
How did you managed to install 2019.3.8f?

I'm trying to investigate this. I was able to install 2019.4.21f via Unity Hub, but cannot install this exact version. As I read [here](https://forum.unity.com/threads/opening-unityhub-urls-with-unity-hub-on-linux.971787/), Unity Hub on Linux doesn't accept `unityhub://` urls.

I try this configuration:
- Unity 2019.4.21f
- VS Code 1.54.1
- mono 6.12.0.122
- Version of the debugger tool: 3.0.11 ( my ["preview" fork](https://marketplace.visualstudio.com/items?itemName=deitry.unity-debug))
- Operating System and version: Kubuntu 18.04

Also I tried to run that gist to see what it lists to me and here is what I got:
```bash
$ mono GetProcesses.exe | grep -i unity 
Process ID 1783 Name 'unityhub-bin'
Process ID 1786 Name 'UnityHub.AppImage'
Process ID 1797 Name 'unityhub-bin'
Process ID 1799 Name 'unityhub-bin'
Process ID 1826 Name 'unityhub-bin'
Process ID 1830 Name 'unityhub-bin'
Process ID 1853 Name 'unityhub-bin'
Process ID 8441 Name 'Unity'
Process ID 8443 Name 'Unity.Licensing.Client'
Process ID 8462 Name 'UnityPackageManager'
Process ID 8547 Name 'UnityShaderCompiler'
Process ID 8567 Name 'UnityHelper'
Process ID 8592 Name 'UnityShaderCompiler'
Process ID 8600 Name 'UnityShaderCompiler'
```

There is no Unity Editor as well yet I got one on `Unity Attach Debugger` command:
![Screenshot_20210306_010023](https://user-images.githubusercontent.com/6714840/110178859-1f0a0f80-7e18-11eb-84ab-3d2cfa65e11a.png)
and also able to connect automatically when starting `Unity Editor` debug configuration.

---
As I can see [here](https://github.com/Unity-Technologies/MonoDevelop.Debugger.Soft.Unity/blob/7a99cf7c707d1d60e968c42a9aec8a55413e5deb/UnityProcessDiscovery.cs#L172), debugger consider all processes named 'Unity' and 'UnityEditor' as editor processes. I got one in `Process ID 8441 Name 'Unity'` while yours probably is `Process ID 6062 Name 'Unity.app'`. Could you please confirm it by checking `ps -aux | grep -i unity`?

Editor process should have something like that, please note the running program
```
user      8441  7.3  1.6 7334692 538488 ?      Ssl  00:59   2:27 /home/user/Unity/Hub/Editor/2019.4.21f1/Editor/Unity -projectpath /home/user/workspace/unity/test -useHub -hubIPC
```

**Comment by andronio-mf** [comment]
I installed unity 2019.3.8f by starting UnityHub with the appropriate unityhub:// link. It does work correctly on my platform (ubuntu 18.04). I am required to use that version because my whole team is using it, so I cannot change it. We want to use the same tools.

Thanks for your reply. I finally figured out what the problem is. I need to change the cache folders for Unity as my main drive is almost full. So I renamed `Unity` to `Unity.app` and created a new executable script named `Unity` in the Editor folder with the following content:

```
export UPM_NPM_CACHE_PATH="/mnt/data/Caches/unity/npm"
export UPM_CACHE_PATH="/mnt/data/Caches/unity/packages"

/mnt/data/Programs/unity/2019.3.8f1/Editor/Unity.app $@
```

But now the process is named `Unity.app` instead of `Unity`. The extension debug output suggests that you can change the name of the process to look for by changing the name of the configuration in the vscode launch.json file. In fact, if I change my configuration name to "Unity.app", the debug output is:

```
UnityDebug: Searching for Unity process 'Unity.app'
```

So I guess if I want to maintain my cached folders elsewhere, all I need to do is launch Unity manually without UnityHub?

**Comment by deitry** [comment]
So were you able to connect debugger after change configuration in launch.json?

It seems that changing configuration name is most preferable way, though I'm not sure whether it's working as expected. Ideally you should be able to connect to yours `Unity.app` process even when launched from Hub

**Comment by andronio-mf** [comment]
No, I was never able to attach the debugger to a process even changing the configuration name. I think the output message is just a log message that bears no actual meaning to how the process is actually detected.

The way I started my editor now is simply by not using the script I was using before (i.e. letting the hub launch the `Unity` executable), which still leaves me with caches in the wrong place. But I could get around that by starting unity through my script manually instead than going through the hub.
